Avoid an implicit declaration of the malloc function.  Improves C99
compatibility.

Submitted upstream:

  <https://lists.gnu.org/archive/html/emacs-devel/2022-12/msg01020.html>

diff --git a/configure.ac b/configure.ac
index cc390d40b33a1d8f..7be08de740cfc486 100644
--- a/configure.ac
+++ b/configure.ac
@@ -4590,6 +4590,7 @@ AC_CACHE_CHECK([whether signals can be handled on alternate stack],
 	       [emacs_cv_alternate_stack],
   [AC_COMPILE_IFELSE(
      [AC_LANG_PROGRAM([[#include <signal.h>
+			#include <stdlib.h>
 		      ]],
 		      [[stack_t ss;
 			struct sigaction sa;
